src/ui/public/vislib/lib/tilemap_settings.js

```diff
@@ -0,0 +1,135 @@
+import uiModules from 'ui/modules';
+import _ from 'lodash';
+import marked from 'marked';
+import url from 'url';
+
+marked.setOptions({
+  gfm: true, // Github-flavored markdown
+  sanitize: true // Sanitize HTML tags
+});
+
+
+uiModules.get('kibana')
+  .service('tilemapSettings', function ($http, mapsConfig, $sanitize) {
+
+    const attributionFromConfig = $sanitize(marked(mapsConfig.deprecated.config.options.attribution));
+    const optionsFromConfig = _.assign({}, mapsConfig.deprecated.config.options, {attribution: attributionFromConfig});
+
+    /**
+     * Object to read out the map-service configuration
+     *
+     * @type {Object}
+     */
+    const mapSettings = {
```

Comment:
I think a class is appropriate anytime we are building an object that tracks state, has some sort of lifecycle, or is more than just "some related data". It communicates that we are defining the shape of an object, regardless how many times we will be constructed. It also makes things like `this` context a little more clear to less advanced folks.